airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py

```diff
@@ -202,7 +200,6 @@ class Config:
     credentials: Optional[Union[OAuthCredentials, ServiceAccountCredentials]] = Field(
```

Comment:
without the discriminator field you should remove "Optional" here as well.

Reply:
Actually, I'm starting to feel less comfortable with this approach as I'm seeing these objects are gone from required as I remove the discriminator and that feels odd 

<img width="1403" alt="image" src="https://github.com/airbytehq/airbyte/assets/168454423/31088abf-6c81-421a-a011-66997cb28bc6">

Reply:
Actually ignore previous comment, let me do more testing on this.

Reply:
Actually this comment was very useful, thanks @cmm-airbyte